File: arb-rs/src/schema/nba/teams.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TeamProfile {
    #[serde(rename = "Active")]
    pub active: bool,
    
    #[serde(rename = "City")]
    pub city: String,
    
    #[serde(rename = "Conference")]
    pub conference: String,
    
    #[serde(rename = "Division")]
    pub division: String,
    
    #[serde(rename = "GlobalTeamID")]
    pub global_team_id: i32,
    
    #[serde(rename = "HeadCoach")]
    pub head_coach: String,
    
    #[serde(rename = "Key")]
    pub key: String,
    
    #[serde(rename = "LeagueID")]
    pub league_id: i32,
    
    #[serde(rename = "Name")]
    pub name: String,
    
    #[serde(rename = "NbaDotComTeamID")]
    pub nba_dot_com_team_id: i32,
    
    #[serde(rename = "PrimaryColor")]
    pub primary_color: String,
    
    #[serde(rename = "QuaternaryColor")]
    pub quaternary_color: String,
    
    #[serde(rename = "SecondaryColor")]
    pub secondary_color: String,
    
    #[serde(rename = "StadiumID")]
    pub stadium_id: i32,
    
    #[serde(rename = "TeamID")]
    pub team_id: i32,
    
    #[serde(rename = "TertiaryColor")]
    pub tertiary_color: String,
    
    #[serde(rename = "WikipediaLogoUrl")]
    pub wikipedia_logo_url: String,
    
    #[serde(rename = "WikipediaWordMarkUrl")]
    pub wikipedia_word_mark_url: Option<String>,
}

pub type TeamProfiles = Vec<TeamProfile>;

File: arb-www/src/schema/mlb/teams.ts
import { BaseDTO } from '../BaseDTO';

/**
 * MLB Team Profile information
 */
export class MLBTeamProfile extends BaseDTO {
  TeamID!: number;
  Key!: string;
  Active!: boolean;
  City!: string;
  Name!: string;
  StadiumID!: number;
  Conference!: string;
  Division!: string;
  PrimaryColor!: string;
  SecondaryColor!: string;
  TertiaryColor!: string;
  QuaternaryColor!: string;
  WikipediaLogoUrl!: string;
  WikipediaWordMarkUrl!: string;
  GlobalTeamID!: number;
  HeadCoach!: string;
  ConferenceID!: number;
  DivisionID!: number;
}

/**
 * MLB Team Profiles response
 */
export class MLBTeamProfilesResponse extends BaseDTO {
  data!: MLBTeamProfile[];
  league!: string;
}
